storage/storage.go

```diff
@@ -43,13 +43,25 @@ type TupleIterator interface {
 type Writes = []*openfgapb.TupleKey
 type Deletes = []*openfgapb.TupleKey
 
+// ListObjectsFilter represents fields that can be used to additionally constrain/filter
+// relationship tuples when querying them.
+type ListObjectsFilter struct {
+	StoreID    string
+	ObjectType string
+}
+
 // A TupleBackend provides an R/W interface for managing tuples.
 type TupleBackend interface {
 	// Read the set of tuples associated with `store` and `key`, which may be partially filled. A key must specify at
 	// least one of `Object` or `User` (or both), and may also optionally constrain by relation. The caller must be
 	// careful to close the TupleIterator, either by consuming the entire iterator or by closing it.
 	Read(context.Context, string, *openfgapb.TupleKey) (TupleIterator, error)
 
+	// ListObjectsByType returns all the objects of a specific type.
+	// You can assume that the type has already been validated.
+	// The result can't have duplicate elements.
+	ListObjectsByType(ctx context.Context, filter ListObjectsFilter) ([]string, error)
```

Comment:
I think if we're going to do this differently than the proposal described, then at a minimum we should yield an iterator instead of a slice. Yielding an iterator is important here to reduce the memory consumption of downstream components that call this method.

Reply:
Ah the reason i don't return a `tupleIterator` is that i don't have tuples to return, just `object_id`s. I could build a brand new iterator just for this, but it felt overkill

Reply:
I think it's important, because the StreamedListObjects API shouldn't have to have a big memory footprint like this. That was one of the whole reasons for the streaming variation, was to only keep iterators/cursors as it evaluates the objects instead of loading them all into memory.

Reply:
Good point. Implemented the iterators 😄